### PR TITLE
[WebKitBrowser] Improve GLib-API support

### PR DESCRIPTION
**Pull request by emutavchi**
Based on work done by Haseena:
https://github.com/HaseenaSainul/rdkservices/tree/development/webkit-iface-syncup

### CODE REVIEW

**Comment by emutavchi** [comment]
@HaseenaSainul @MFransen69 could you please review?

**Review by HaseenaSainul — CHANGES_REQUESTED** [review]
In general, code changes seems fine for me. 
But I am not happy with Injected bundle glib version keeping outside InjectedBundle folder extension. So I would like to have both webkit version changes under injected bundle itself. there you can create subfolders for both versions with appropriate naming

File: WebKitBrowser/WebKitImplementation.cpp [hunk 1]
```diff
@@ -2125,17 +2201,19 @@ static GSourceFuncs _handlerIntervention =
...
 
-            g_signal_connect(_view, "decide-policy", reinterpret_cast<GCallback>(decidePolicyCallback), nullptr);
+            g_signal_connect(_view, "decide-policy", reinterpret_cast<GCallback>(decidePolicyCallback), this);
             g_signal_connect(_view, "notify::uri", reinterpret_cast<GCallback>(uriChangedCallback), this);
             g_signal_connect(_view, "load-changed", reinterpret_cast<GCallback>(loadChangedCallback), this);
             g_signal_connect(_view, "web-process-terminated", reinterpret_cast<GCallback>(webProcessTerminatedCallback), nullptr);
             g_signal_connect(_view, "close", reinterpret_cast<GCallback>(closeCallback), this);
             g_signal_connect(_view, "permission-request", reinterpret_cast<GCallback>(decidePermissionCallback), nullptr);
             g_signal_connect(_view, "show-notification", reinterpret_cast<GCallback>(showNotificationCallback), this);
+            g_signal_connect(_view, "user-message-received", reinterpret_cast<GCallback>(userMessageReceivedCallback), this);
```
[the remark this answers is not in the document]
Comment (emutavchi) [reply WebKitBrowser/WebKitImplementation.cpp:2217]: Hi @HaseenaSainul, thanks for looking into this.

> So I would like to have both webkit version changes under injected bundle itself. there you can create subfolders for both versions with appropriate naming

not sure what you mean. Injected bundle is part of WebKit (with GLib-API enabled). So, it makes sense to me to move Extension sources out of plugin InjectedBundle.

> Why you have skipped the g_signal_connect for notify::is-web-process-responsive ?

Webkit Glib API triggers it after the web process crash, which was a bit confusing. So, I removed it at some point and probably forgot about it. I can bring it back.

[the remark this answers is not in the document]
Comment (emutavchi) [reply WebKitBrowser/WebKitImplementation.cpp:2218]: @HaseenaSainul  I've brought back handling of notify::is-web-process-responsive, could you please review?

**Review by MFransen69 — CHANGES_REQUESTED** [review]

File: WebKitBrowser/CMakeLists.txt [hunk 2]
```diff
@@ -179,8 +179,12 @@ install(
     DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/${STORAGE_DIRECTORY}/plugins
     COMPONENT ${MODULE_NAME})
 
-# also build injected bundle
-add_subdirectory(InjectedBundle)
+if(WEBKIT_GLIB_API)
+    add_subdirectory(Extension)
```
[the remark this answers is not in the document]
Comment (emutavchi) [reply WebKitBrowser/CMakeLists.txt:183]: Hi @MFransen69, thanks for review.

My understanding is that GLib API doesn't allow user-provided injected bundles, and that was the reason for switching to extension API. there are other things that are currently not supported with extension API and need to be added later.

I did address your concern with the configuration.

[the remark this answers is not in the document]
Comment (HaseenaSainul) [reply WebKitBrowser/CMakeLists.txt:183]: Then where will be used libWPEExtension.so ? In metro build we are loading GLIB version of injected bundle with wpe-2.28 version and it is working.

**Review by MFransen69 — APPROVED** [review]

**Review by HaseenaSainul — APPROVED** [review]

**Review by goruklu — APPROVED** [review]

**Pull request merged**